src/Homepage/index.js

```diff
@@ -163,6 +164,54 @@ class Homepage extends React.Component {
   }
 
   async componentDidMount() {
+    if (this.props.currentLocation.latLng !== undefined) {
+      this.hidePageLoader();
+    }
+    else {
+      if('geolocation' in navigator) {
+        navigator.geolocation.getCurrentPosition(
```

Comment:
This looks great, but I think we should move this to `src/App.js` instead. Since the concept of current location is shared across pages, I've moved the implementation details of ip/geo to App.js, and rely on `currentLocation` (with accuracy as high/low) in the page components. The loader can be passed as a prop to this component

Reply:
I might need some clarification.
1. Moving this to `src/App.js`:  do you mean to make it a function that gets passed down and called by HomePage, or to make it a function that's called when the App component is created? 
2. Loader passed down as a prop: so, passing down `isPageLoading` as a prop instead having it as a state in HomePage?

Reply:
Thanks, @xsihe
>Moving this to src/App.js: do you mean to make it a function that gets passed down and called by HomePage, or to make it a function that's called when the App component is created?

When the App component is created/mounted.

>Loader passed down as a prop: so, passing down isPageLoading as a prop instead having it as a state in HomePage?

Yes. It could be more explicitly called "isCurrentLocationLoading" or something like that.